### PR TITLE
Add ZeroSSL

### PR DESCRIPTION
**Pull request by Moudoux**
Adds ZeroSSL endpoints to types.rs

### CODE REVIEW

File: src/types.rs [hunk 1]
```diff
@@ -422,8 +422,24 @@ impl LetsEncrypt {
     /// Get the directory URL for the given Let's Encrypt server
     pub const fn url(&self) -> &'static str {
         match self {
-            LetsEncrypt::Production => "https://acme-v02.api.letsencrypt.org/directory",
-            LetsEncrypt::Staging => "https://acme-staging-v02.api.letsencrypt.org/directory",
+            Self::Production => "https://acme-v02.api.letsencrypt.org/directory",
+            Self::Staging => "https://acme-staging-v02.api.letsencrypt.org/directory",
+        }
```
Comment (djc) [inline src/types.rs:427]: I'd like this change (which is mostly unrelated) to be in a separate commit.

File: src/types.rs [hunk 2]
```diff
@@ -422,8 +422,24 @@ impl LetsEncrypt {
...
+    }
+}
+
+/// ZeroSSL ACME only supports production at the moment
+#[allow(missing_docs)]
+#[derive(Clone, Copy, Debug)]
+pub enum ZeroSSL {
```
Comment (djc) [inline src/types.rs:434]: This type should be called `ZeroSsl` to match the Rust API naming guidelines.

**Review by djc — COMMENTED** [review]
This mostly looks good to me. Can you squash most of the intermediate commits and isolate the change to the `LetsEncrypt::url()` implementation in a separate commit?

**Comment by Moudoux** [comment]
Done, great library btw

**Comment by djc** [comment]
Thanks -- happy you like the library! Please open issues for any other feedback you might have.
